test/replutil.jl

```diff
@@ -3,42 +3,25 @@
 # For curmod_*
 include("testenv.jl")
 
-function test_have_color(buf, color, no_color)
-    if Base.have_color
-        @test String(take!(buf)) == color
-    else
-        @test String(take!(buf)) == no_color
-    end
-end
-
 cfile = " at $(@__FILE__):"
 c1line = @__LINE__() + 1
 method_c1(x::Float64, s::AbstractString...) = true
 
 buf = IOBuffer()
 Base.show_method_candidates(buf, Base.MethodError(method_c1,(1, 1, "")))
-no_color = "\nClosest candidates are:\n  method_c1(!Matched::Float64, !Matched::AbstractString...)$cfile$c1line"
+@test String(take!(buf)) == "\nClosest candidates are:\n  method_c1(!Matched::Float64, !Matched::AbstractString...)$cfile$c1line"
 @test length(methods(method_c1)) <= 3 # because of '...' in candidate printing
-test_have_color(buf,
-                "\e[0m\nClosest candidates are:\n  method_c1(\e[1m\e[31m::Float64\e[0m, \e[1m\e[31m::AbstractString...\e[0m)$cfile$c1line\e[0m",
```

Comment:
might be nice to keep (some of?) these, now that we can actually test both code path properly:

```julia
Base.show_method_candidates(IOContext(buf, :color => true), Base.MethodError(method_c1,(1, 1, "")))
@test String(take!(buf)) == "\e[0m\nClosest candidates are:\n  method_c1(\e[1m\e[31m::Float64\e[0m, \e[1m\e[31m::AbstractString...\e[0m)$cfile$c1line\e[0m"
```

Reply:
I think color codes in warnings are tested elsewhere....

Reply:
But sure, I'll add a few back in with color enabled.

Reply:
Interestingly enough, the `have_color` testcase was totally wrong, since at least Julia 0.6.   It seems that this hasn't been tested in a long while.

Reply:
I've added a test.

Reply:
Proving once again that bitrot is inevitable for code that doesn't actually run!